lib/puppet/provider/group/groupadd.rb

```diff
@@ -82,4 +83,27 @@ def addcmd
     cmd << @resource[:name]
     cmd
   end
+
+  def modifycmd(param, value)
+    if @resource.forcelocal?
+      cmd = [command(:localmodify)]
```

Comment:
It looks like the use of `forcelocal` in `addcmd` is also setting a custom environment (line 71). I'd suggest doing the same here for consistency.

Reply:
Should all the forcelocal commands use the custom environment?  This would be `delete` for groupadd and several more for useradd.

Reply:
I *think* so, yes. All this does is add `{"LIBUSER_CONF"=>"/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/libuser.conf"}` to `@custom_environment` (or whatever pathing is appropriate for your puppet version). This will ensure that environment variable is set for the duration of the command that is being run so that we isolate which libuser.conf is being picked up when we are forcing local. Since this config impacts all the `l{user,group}*` commands, I think we do need to set the custom environment everywhere.